cache-invalidation/server.sh

```diff
@@ -0,0 +1,179 @@
+
+	    mkdir build
+	    pushd build
+
+	    cmake -DCMAKE_BUILD_TYPE=Maintainer -DBUILD_CONFIG=everything ../src
+	    make dist
+	    rpmbuild -ta --define "_srcrpmdir $PWD" --define "_rpmdir $PWD" *.tar.gz
+	    rpm_arch=$(rpm -E '%{_arch}')
+	    ganesha_version=$(rpm -q --qf '%{VERSION}-%{RELEASE}' -p *.src.rpm)
+	    if [ -e ${rpm_arch}/libntirpc-devel*.rpm ]; then
+		    ntirpc_version=$(rpm -q --qf '%{VERSION}-%{RELEASE}' -p ${rpm_arch}/libntirpc-devel*.rpm)
+		    ntirpc_rpm=${rpm_arch}/libntirpc-${ntirpc_version}.${rpm_arch}.rpm
+	    fi
+	    yum -y install ${ntirpc_rpm} ${rpm_arch}/nfs-ganesha-{,gluster-}${ganesha_version}.${rpm_arch}.rpm
+
+	    # start nfs-ganesha service with an empty configuration
+	    > /etc/ganesha/ganesha.conf
+	    if ! systemctl start nfs-ganesha
+	    then
+		    echo "+++ systemctl status nfs-ganesha.service +++"
+		    systemctl status nfs-ganesha.service
+		    echo "+++ journalctl -xe +++"
+		    journalctl -xe
+		    exit 1
+	    fi
+    fi
+
+    # TODO: open only the ports needed?
+    # disable the firewall, otherwise the client can not connect
+    systemctl stop firewalld || service iptables stop
+
+    # install and start gluster
+    yum -y install glusterfs-server
+    systemctl start glusterd
+    mkdir -p /bricks/${GLUSTER_VOLUME}
+fi
+
+if [ $2 -eq 0 ]
+then
+    # create and start gluster volume
+    gluster peer probe $4
+    gluster volume create ${GLUSTER_VOLUME} \
+	replica 2 \
+	$(hostname --fqdn):/bricks/${GLUSTER_VOLUME}/b{1,2} $4:/bricks/${GLUSTER_VOLUME}/b{1,2} force
+
+    gluster volume start ${GLUSTER_VOLUME} force
+    
+    # cache-invalidation on for the volume
+    gluster v set ${GLUSTER_VOLUME} features.cache-invalidation on
+fi
+
+if [ $3 -eq 0 ]
+then
+    # Export the volume
+    mkdir -p /usr/libexec/ganesha
+    cd /usr/libexec/ganesha
+    yum -y install wget
+    wget https://raw.githubusercontent.com/gluster/glusterfs/release-3.10/extras/ganesha/scripts/create-export-ganesha.sh
+    wget https://raw.githubusercontent.com/gluster/glusterfs/release-3.10/extras/ganesha/scripts/dbus-send.sh
+    chmod 755 create-export-ganesha.sh dbus-send.sh
+
+    /usr/libexec/ganesha/create-export-ganesha.sh /etc/ganesha on ${GLUSTER_VOLUME}
+    /usr/libexec/ganesha/dbus-send.sh /etc/ganesha on ${GLUSTER_VOLUME}
+
+    # wait till server comes out of grace period
+    sleep 90
+
+    # basic check if the export is available, some debugging if not
+    if ! showmount -e | grep -q -w -e "${GLUSTER_VOLUME}"
+    then
+	    echo "+++ /var/log/ganesha.log +++"
+	    cat /var/log/ganesha.log
+	    echo
+	    echo "+++ /etc/ganesha/ganesha.conf +++"
+	    grep --with-filename -e '' /etc/ganesha/ganesha.conf
+	    echo
+	    echo "+++ /etc/ganesha/exports/*.conf +++"
+	    grep --with-filename -e '' /etc/ganesha/exports/*.conf
+	    echo
+	    echo "Export ${GLUSTER_VOLUME} is not available"
+	    exit 1
+    fi
+
+    #Enabling ACL for the volume if ENABLE_ACL param is set to True
```

Comment:
Please move these checks prior to line:146

Reply:
This is done as https://github.com/nfs-ganesha/ci-tests/blob/centos-ci/common-scripts/basic-gluster.sh which is used by other ci-tests. Should we change there also?